[Bug]: build is failing for new Otter library

### Package name

core, workspace

### Package version

10.2.1

### Reproduction steps

```shell
npm create @o3r otter-project
cd otter-project
ng g library somelib
npm run ng build somelib
```

### Current result

```shell

------------------------------------------------------------------------------
Building entry point 'somelib'
------------------------------------------------------------------------------
✖ Compiling with Angular sources in Ivy partial compilation mode.
error TS5090: Non-relative paths are not allowed when 'baseUrl' is not set. Did you forget a leading './'?
```


Adding `baseUrl` by hand leads to another error:

`Dependency @o3r/core must be explicitly allowed using the "allowedNonPeerDependencies" option.`


### Expected result

Build success

### Additional comments

The second error comes from the fact that `@o3r/core` is added as `dependency` and not `devDependency` on the generated library.

--- a/packages/@o3r/workspace/schematics/library/index.ts
+++ b/packages/@o3r/workspace/schematics/library/index.ts
@@ -46,7 +46,7 @@ function generateModuleFn(options: NgGenerateModuleSchema): Rule {
         inManifest: [
           {
             range: `${options.exactO3rVersion ? '' : '~'}${ownPackageJsonContent.version!}`,
-            types: [NodeDependencyType.Default]
+            types: [NodeDependencyType.Dev]
           }
         ],
         ngAddOptions: { exactO3rVersion: options.exactO3rVersion }

--- a/packages/@o3r/workspace/schematics/library/rules/rules.ng.ts
+++ b/packages/@o3r/workspace/schematics/library/rules/rules.ng.ts
@@ -48,6 +48,8 @@ export function ngGenerateModule(options: NgGenerateModuleSchema & { targetPath:
         const genPackageJsonPath = path.posix.join(options.targetPath, 'package.json');
         const packageJson = t.readJson(genPackageJsonPath) as PackageJson;
         packageJson.name = options.packageJsonName;
+        packageJson.scripts ||= {};
+        packageJson.scripts.ng = 'ng';
         t.overwrite(genPackageJsonPath, JSON.stringify(packageJson, null, 2));
         return t;
       }
@@ -62,37 +64,25 @@ export function ngGenerateModule(options: NgGenerateModuleSchema & { targetPath:
   const updateTsConfigFiles: Rule = (tree, context) => {
     const tsconfigBase = findConfigFileRelativePath(tree, ['tsconfig.base.json', 'tsconfig.json'], '/');
     const tsconfigBuild = findConfigFileRelativePath(tree, ['tsconfig.build.json'], '/');
-    if (tsconfigBase) {
-      const configFile = tree.readJson(tsconfigBase) as TsConfigJson;
-      if (configFile?.compilerOptions?.paths) {
-        configFile.compilerOptions.paths = Object.fromEntries(
-          Object.entries(configFile.compilerOptions.paths).filter(([pathName, _]) => pathName !== options.name));
-        configFile.compilerOptions.paths[options.packageJsonName] = [
-          path.posix.join(relativeTargetPath, 'dist'),
-          path.posix.join(relativeTargetPath, 'src', 'public-api')
-        ];
-        tree.overwrite(tsconfigBase, JSON.stringify(configFile, null, 2));
+    [tsconfigBase, tsconfigBuild].forEach(tsconfigPath => {
+      if (tsconfigPath) {
+        const configFile = tree.readJson(tsconfigPath) as TsConfigJson;
+        if (configFile?.compilerOptions?.paths) {
+          configFile.compilerOptions.baseUrl = '.';
+          configFile.compilerOptions.paths = Object.fromEntries(
+            Object.entries(configFile.compilerOptions.paths).filter(([pathName, _]) => pathName !== options.name));
+          configFile.compilerOptions.paths[options.packageJsonName] = [
+            path.posix.join(relativeTargetPath, 'dist'),
+            path.posix.join(relativeTargetPath, 'src', 'public-api')
+          ];
+          tree.overwrite(tsconfigPath, JSON.stringify(configFile, null, 2));
+        } else {
+          context.logger.warn(`${tsconfigPath} does not contain path mapping, the edition will be skipped`);
+        }
       } else {
-        context.logger.warn(`${tsconfigBase} does not contain path mapping, the edition will be skipped`);
+        context.logger.warn(`No TsConfig file found at ${tsconfigPath}`);
       }
-    } else {
-      context.logger.warn('No base TsConfig file found');
-    }
-
-    if (tsconfigBuild) {
-      const configFile = tree.readJson(tsconfigBuild) as TsConfigJson;
-      if (configFile?.compilerOptions?.paths) {
-        configFile.compilerOptions.paths = Object.fromEntries(
-          Object.entries(configFile.compilerOptions.paths).filter(([pathName, _]) => pathName !== options.name));
-        configFile.compilerOptions.paths[options.packageJsonName] = [
-          path.posix.join(relativeTargetPath, 'dist'),
-          path.posix.join(relativeTargetPath, 'src', 'public-api')
-        ];
-        tree.overwrite(tsconfigBuild, JSON.stringify(configFile, null, 2));
-      } else {
-        context.logger.warn(`${tsconfigBuild} does not contain path mapping, the edition will be skipped`);
-      }
-    }
+    });
   };
 
   const ngCliUpdate: Rule = (tree, context) => {

